l10n_it_amount_to_text/models/res_currency.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2020 Ecosoft Co., Ltd (http://ecosoft.co.th)
+# Copyright 2022 Sergio Zanchetta (Associazione PNLUG - Gruppo Odoo)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from num2words import num2words
+
+from odoo import models, tools
+
+CURRENCY_NAME = {"Dollars": "dollari", "Euros": "euro", "Cents": "centesimi"}
+
+
+class Currency(models.Model):
+    _inherit = "res.currency"
+
+    def _convert_currency_name_hook(self, label):
+        """Hooks to add currency translation.
+        Currently supported currencies are USD and EUR"""
+        return CURRENCY_NAME[label]
+
+    def amount_to_text(self, amount):
+        self.ensure_one()
+        lang_code = self.env.context.get("lang") or False
+        if lang_code != "it_IT":
+            return super().amount_to_text(amount)
+
+        formatted = "%.{}f".format(self.decimal_places) % amount
+        parts = formatted.partition(".")
+        integer_value = int(parts[0])
+        fractional_value = int(parts[2] or 0)
+        lang = (
+            self.env["res.lang"]
+            .with_context(active_test=False)
+            .search([("code", "=", lang_code)])
+        )
+
+        # use num2words native 'currency' option to handle exceptions.
+        # eg. 'uno euro' -> 'un euro', 'uno centesimi' -> 'un centesimo'"""
+        if self.name == "EUR":
+            try:
+                return num2words(amount, to="currency", lang=lang.iso_code)
+            except NotImplementedError:
+                return num2words(amount, to="currency", lang="en")
+
+        currency_unit_label = self._convert_currency_name_hook(self.currency_unit_label)
+        amount_words = tools.ustr("{amt_value} {amt_word}").format(
+            amt_value=num2words(integer_value, lang=lang.iso_code),
```

Comment:
Qui e sotto, come mai qui non viene usato il fallback su `lang="en"`?
Anche l'aggiunta di `to="currency"` penso serva, in modo da gestire le eccezioni (tipo "una sterlina"), che ne pensi?

Reply:
Lo scopo originario della PR era di fornire solo una base (con supporto a €) che potesse poi essere estesa successivamente.

Ad ogni modo ho colto l'occasione per rivedere la PR in modo che gestisca la forma singolare di tutte le valute introdotte da https://github.com/savoirfairelinux/num2words/pull/434 in num2words v. 0.5.12. (euro, dollaro, sterlina e yuan).
Per tutte le altre valute la PR adesso gestisce il passaggio maiuscolo/minuscolo ma non la forma singolare.